lib/hammer_cli_foreman/api/connection.rb

```diff
@@ -1,22 +1,24 @@
 require 'hammer_cli_foreman/api/session_authenticator_wrapper'
 require 'hammer_cli_foreman/api/interactive_basic_auth'
+require 'hammer_cli_foreman/api/oauth/interactive_keycloak_auth_via_2fa'
+require 'hammer_cli_foreman/api/oauth/interactive_keycloak_auth'
 require 'hammer_cli_foreman/api/void_auth'
 
 module HammerCLIForeman
   module Api
     class Connection < HammerCLI::Apipie::ApiConnection
       attr_reader :authenticator
 
-      def initialize(settings, logger = nil, locale = nil)
-        default_params = build_default_params(settings, logger, locale)
+      def initialize(settings, logger = nil, locale = nil, auth_type = AuthTypes[:basic_auth])
```

Comment:
Can we use constants instead of the hash for auth_type? Something like AUTH_TYPE_BASIC, AUTH_TYPE_OAUTH_BASIC, AUTH_TYPE_OAUTH_2FA would be more readable.